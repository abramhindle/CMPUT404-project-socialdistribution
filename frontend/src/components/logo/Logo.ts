import { FASTElement, observable } from "@microsoft/fast-element";

export class Logo extends FASTElement {
    @observable
    public logoUrl = require('../../assets/images/LogoIcon.png').default;
}