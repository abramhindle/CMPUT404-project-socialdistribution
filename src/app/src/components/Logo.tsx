import React, { useState } from 'react';
import styled from 'styled-components';

const LogoContainer = styled.div`
  text-align: center;
  height: ${window.innerHeight}px;
  font-family: Avenir Next Light;
  font-size: 200px;

  &::first-letter {
    font-size: 300px;
  }
`;

export default function Logo() {
  return <LogoContainer>Website.</LogoContainer>;
}
