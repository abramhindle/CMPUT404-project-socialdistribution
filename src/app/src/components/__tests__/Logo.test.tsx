import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';

import Logo from '../Logo';

let container: any = null;
//issues with container type and its methods

beforeEach(() => {
  // setup a DOM element as a render target
  container = document?.createElement('div');
  document?.body?.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container?.remove();
  container = null;
});

describe('Logo suite', () => {
  it('Renders correctly', () => {
    const tree = renderer.create(<Logo />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Logo text', () => {
    render(<Logo />, container);
    expect(container.textContent).toBe('Website.');
  });
});
