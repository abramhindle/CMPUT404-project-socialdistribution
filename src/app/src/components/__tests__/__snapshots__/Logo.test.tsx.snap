// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo suite Renders correctly 1`] = `
<div
  className="sc-bdvvtL bxQDqD"
>
  Website.
</div>
`;
